api.py

```
@@ -0,0 +1,283 @@
+    if os.path.exists(TEMP_UPLOAD_DIR):
+        try:
+            shutil.rmtree(TEMP_UPLOAD_DIR)
+            print(f"Cleaned up temporary upload directory: {TEMP_UPLOAD_DIR}")
+        except Exception as e:
+            print(f"Error cleaning up temporary upload directory {TEMP_UPLOAD_DIR}: {e}")
+    print("FastAPI application shutdown complete.")
+
+# --- API Endpoints ---
+
+@app.post("/ingest_xml/", response_model=IngestResponse, tags=["Data Ingestion"])
+async def ingest_xml_file(file: UploadFile = File(...)):
```

Ruff (0.11.9)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)